src/org/zaproxy/zap/extension/ascanrulesAlpha/HttpsAsHttpScanner.java

```diff
@@ -1,172 +1,172 @@
+		return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+	}
+
+	@Override
+	public String getSolution() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "soln");
+	}
+
+	@Override
+	public String getReference() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "refs");
+	}
+
+	@Override
+	public String[] getDependency() {
+		return null;
+	}
+
+	@Override
+	public int getCategory() {
+		return Category.MISC;
+	}
+	
+	@Override
+	public int getRisk() {
+		return Alert.RISK_LOW;
+	}
+
+	@Override
+	public int getCweId() {
+		return 311; // CWE-311: Missing Encryption of Sensitive Data
+	}
+
+	@Override
+	public int getWascId() {
+		return 4; // WASC-04: Insufficient Transport Layer Protection
+	}
+
+	@Override
+	public void init() {
+
+	}
+
+	@Override
+	public void scan() {
+		
+		if (!getBaseMsg().getRequestHeader().isSecure()) { //Base request isn't HTTPS
+			if (log.isDebugEnabled()) { 
+				log.debug ("The original request was not HTTPS, so there is not much point in looking further.");
+			}
+			return;	
+		}
+		
+		int originalStatusCode = getBaseMsg().getResponseHeader().getStatusCode();
+		if (originalStatusCode == HttpStatusCode.NOT_FOUND || originalStatusCode == 0) {
+			if (log.isDebugEnabled()) {
+				log.debug ("The original request was not successfuly completed (status = "+ originalStatusCode +"), so there is not much point in looking further.");
+			}
+			return;		
+		}
+		
+		if (log.isDebugEnabled()) {
+			log.debug("Checking if " + getBaseMsg().getRequestHeader().getURI() + " is available via HTTP.");
+		}
+		
+		HttpMessage newRequest = getNewMsg();
+		
+		try{
+			newRequest.getRequestHeader().setSecure(false); //https becomes http
+			if (log.isDebugEnabled()) {
+				log.debug("**"+newRequest.getRequestHeader().getURI());
+			}
+		} catch (URIException e) {
+			log.error("Error creating HTTP URL from HTTPS URL:", e);
+			return; 
+		}
+
+		//Check if the user stopped things. One request per URL so check before sending the request
+		if (isStop()) { 
+			if (log.isDebugEnabled()) {
+				log.debug("Scanner "+getName()+" Stopping.");
+			}
+			return; 
+		}
+		
+		try {
+			sendAndReceive(newRequest, false);
+		} catch (IOException e) {
+			log.error("Error scanning a request via HTTP when the original was HTTPS:", e);
+			return; 
+		}
+		
+		if (newRequest.getResponseHeader().getStatusCode() == HttpStatusCode.OK) { // 200 Success
+
+			String newUri = newRequest.getRequestHeader().getURI().toString();
+			
+			bingo(getRisk(), //Risk
+					Alert.CONFIDENCE_MEDIUM, //Confidence/Reliability
+					getName(), //Name
+					getDescription(), //Description
+					getBaseMsg().getRequestHeader().getURI().toString(), //Original URI
+					null, //Param
+					"", //Attack
+					Constant.messages.getString(MESSAGE_PREFIX+"otherinfo", newUri), //OtherInfo 
+					getSolution(), //Solution
+					newUri, //Evidence
+					getCweId(), //CWE ID
+					getWascId(), //WASC ID
+					newRequest); //HTTPMessage
+		}
+	}
+	
+}
```

Comment:
Did you change all the line endings?

Reply:
That happened while restoring the file, it will be addressed in following commit.

Reply:
K

Reply:
@AnamikaD further changes coming?